core/src/main/java/org/infinispan/container/impl/BoundedSegmentedDataContainer.java

```diff
@@ -103,13 +96,35 @@ protected void computeEntryWritten(K key, InternalCacheEntry<K, V> value) {
 
    @Override
    protected void computeEntryRemoved(K key, InternalCacheEntry<K, V> value) {
-      int segment = getSegmentForKey(key);
+      computeEntryRemoved(getSegmentForKey(key), key, value);
+   }
+
+   protected void computeEntryRemoved(int segment, K key, InternalCacheEntry<K, V> value) {
       ConcurrentMap<K, InternalCacheEntry<K, V>> map = BoundedSegmentedDataContainer.super.getMapForSegment(segment);
       if (map != null) {
          map.remove(key, value);
       }
    }
 
+   @Override
+   protected void putEntryInMap(PeekableTouchableMap<K, V> map, int segment, K key, InternalCacheEntry<K, V> ice) {
+      map.computeIfAbsent(key, k -> {
```

Comment:
is it intentional to use `computeIfAbsent`?

Reply:
No, should really be `compute`.